### PR TITLE
[drogon] fix: patch DrogonConfig.cmake.in to find hiredis

### PR DESCRIPTION
**Pull request by hexorer**
In #32694 merge, the `Hiredis_FOUND` variable was removed mistakenly, which caused `cmake/DrogonConfig.cmake.in` to be configured improperly. resulting in error in projects that use drogon port with redis feature to be required to `find_package(hiredis)` in their cmake file before linking to `Drogon::Drogon`.

In order to fix this, the `drogon_config.patch`, which was covering similar issue with postgres, was updated to also cover `hiredis` case.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] ~~SHA512s are updated for each updated download~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by cthulhu-irl** [comment]
@an-tao I would appreciate it, if you would check the patches from #32694 and this merge to see whether it doesn't affect the expected behavior of drogon library.

File: ports/drogon/drogon_config.patch [hunk 1]
```diff
@@ -11,3 +11,14 @@ index a21122a..6367259 100644
...
  if(@SQLite3_FOUND@)
  find_dependency(SQLite3)
+@@ -37,8 +37,8 @@ endif()
+ if(@COZ-PROFILER_FOUND@)
+ find_dependency(coz-profiler)
+ endif()
+-if(@Hiredis_FOUND@)
+-find_dependency(Hiredis)
++if(@hiredis_FOUND@)
++find_dependency(hiredis)
```
Comment (BillyONeal) [inline ports/drogon/drogon_config.patch:21]:
```suggestion
+find_dependency(hiredis CONFIG)
```

**Review by BillyONeal — DISMISSED** [review]
LGTM except the missing CONFIG

File: ports/drogon/drogon_config.patch (hunk 1 above)
Comment (cthulhu-irl) [reply ports/drogon/drogon_config.patch]: done.

File: ports/drogon/vcpkg_redis.patch [hunk 2]
```diff
@@ -34,6 +34,7 @@ index 4406362..62d7f50 100755
 -
 -if (NOT Hiredis_FOUND)
 +    find_package(hiredis CONFIG REQUIRED)
++    set(Hiredis_FOUND TRUE)
```
Comment (dg0yt) [inline ports/drogon/vcpkg_redis.patch:37]:
~~~cmake
find_package(Hiredis NAMES hiredis REQUIRED)
~~~
would take care of setting `Hiredis_FOUND`.

File: ports/drogon/drogon_config.patch [hunk 3]
```diff
@@ -11,3 +11,14 @@ index a21122a..6367259 100644
  endif()
  if(@SQLite3_FOUND@)
  find_dependency(SQLite3)
+@@ -37,8 +37,8 @@ endif()
+ if(@COZ-PROFILER_FOUND@)
+ find_dependency(coz-profiler)
+ endif()
+-if(@Hiredis_FOUND@)
+-find_dependency(Hiredis)
++if(@hiredis_FOUND@)
```
Comment (dg0yt) [inline ports/drogon/drogon_config.patch:20]: The `if` line doesn't have to change now that `Hiredis_FOUND` is set.

**Review by dg0yt — COMMENTED** [review]
IMO `vcpkg_redis.patch` is already too intrusive before this PR, and the use of hiredis config is split over two patch files...

File: ports/drogon/vcpkg_redis.patch (hunk 2 above)
Comment (cthulhu-irl) [reply ports/drogon/vcpkg_redis.patch:37]: There is a `FindHiredis.cmake` in module path. will this for sure be caught by vcpkg find_package override?

**Comment by cthulhu-irl** [comment]
> IMO `vcpkg_redis.patch` is already too intrusive before this PR, and the use of hiredis config is split over two patch files...

I agree. I've aggregated them into a single `deps_redis.patch` for now. I don't like it that multiple patches are modifying the same file too.

This PR is just to solve what my last PR broke. this pattern has been repeated a few more times. I'll make a definitive PR to resolve them all so all features could be used in manifest mode, and I'll tidy up the patches in that PR.

**Review by dg0yt — APPROVED** [review]
LGTM (but only checked on GH). Thanks!

**Comment by Adela0814** [comment]
All feature tested pass on the following triplets:

- x64-windows
- x64-windows-static
- x86-windows

**Review by Adela0814 — APPROVED** [review]

**Comment by cthulhu-irl** [comment]
> Thanks for the fix and sorry I missed it!

@BillyONeal sorry I didn't test it thoroughly last time.

**Comment by BillyONeal** [comment]
No problem!

**Pull request merged**